rules.d/60-net-dhcp.rules

```diff
@@ -0,0 +1,15 @@
+# do not edit this file, it will be overwritten on update
+
+ACTION=="remove", GOTO="net_dhcp_end"
+SUBSYSTEM!="net", GOTO="net_dhcp_end"
+
+# Network interfaces requiring DHCPOFFER messages to be broadcast
+# must set ID_NET_DHCP_BROADCAST to "1". This property will be
+# checked by the networkd DHCP4 client to set the DHCP option
+
+# s390 ccwgroup interfaces in layer3 mode need broadcast DHCPOFFER
+# which is the case if the layer2 attribute of the corresponding
+# qeth device is "0"
+SUBSYSTEMS=="ccwgroup", DRIVERS=="qeth", ATTRS{layer2}=="0", ENV{ID_NET_DHCP_BROADCAST}="1"
```

Comment:
Can we please move this after 80-net-setup-link.rules, by calling this 81-net-dhcp.rules and then use

```suggestion
ID_NET_DRIVER=="qeth_l3", ENV{ID_NET_DHCP_BROADCAST}="1"
```
which is closer to what https://github.com/systemd/systemd/pull/18829/files did.

ID_NET_DRIVER is an existing value that is generated for all interfaces, and hence it is slightly quicker way to compare and identify layer3 qeth device, without any extra lookups or comparisons on any nic / platform.

And then the commit message becomes also cleaner / easier to parse. "Change qeth_l3 devices DHCP Broadcast default value to true, as they typically need it."

Reply:
how is the "qeth_l3" driver loaded at boot? does it have modalias lines? if so, then the modalias is what should be used as hwdb key and the explicit driver match shouldn't be used in the udev rule

Reply:
@poettering as i have mentioned before there are no modaliases. Hence hwdb approach cannot be used. The same ccw id devices look the same, and one can bind the same ccw qeth device as either qeth_l2 or qeth_l3 driver, depending on the underlying mainframe config.

Reply:
As to how is it loaded => the answer is, always manually. There is import of data from HMC that s390-tools acts upon, and/or one can manually load devices, or use chzdev to store those things on disk.

Reply:
ack for using the string from drvinfo (ie what's called ID_NET_DRIVER here), feels cleaner than poking at custom sysfs attributes.